ui/components/multichain/network-manager/hooks/useNetworkManagerState.ts

```diff
@@ -0,0 +1,223 @@
+
+          // Check if the network is NOT a built-in network or featured RPC
+          const isBuiltInNetwork = Object.values(BUILT_IN_NETWORKS).some(
+            (builtInNetwork) => builtInNetwork.chainId === hexChainId,
+          );
+          const isFeaturedRpc = FEATURED_RPCS.some(
+            (featuredRpc) => featuredRpc.chainId === hexChainId,
+          );
+
+          const isMultichainProviderConfig = Object.values(
+            MultichainNetworks,
+          ).some(
+            (multichainNetwork) =>
+              multichainNetwork === network.chainId ||
+              (network.isEvm
+                ? convertCaipToHexChainId(network.chainId)
+                : network.chainId) === multichainNetwork,
+          );
+
+          const shouldInclude =
+            !isBuiltInNetwork && !isFeaturedRpc && !isMultichainProviderConfig;
+
+          if (shouldInclude || isTest) {
+            (isTest ? testnetsList : nonTestnetsList)[chainId] = network;
+          }
+
+          return [nonTestnetsList, testnetsList];
+        },
+        [
+          {} as Record<string, MultichainNetworkConfiguration>,
+          {} as Record<string, MultichainNetworkConfiguration>,
+        ],
+      ),
+    [multichainNetworks],
+  );
+
+  const isNetworkInDefaultNetworkTab = useCallback(
+    (network: MultichainNetworkConfiguration) => {
+      const networkChainId = network.chainId; // eip155:59144
+      // Convert CAIP format to hex format for comparison
+      const hexChainId = network.isEvm
+        ? convertCaipToHexChainId(networkChainId)
+        : networkChainId;
+
+      // Only show networks if they are built-in networks or featured RPCs
+      const isBuiltInNetwork = Object.values(BUILT_IN_NETWORKS).some(
+        (builtInNetwork) => builtInNetwork.chainId === hexChainId,
+      );
+
+      const isFeaturedRpc = FEATURED_RPCS.some(
+        (featuredRpc) => featuredRpc.chainId === hexChainId,
+      );
+
+      const isMultichainProviderConfig = Object.values(MultichainNetworks).some(
+        (multichainNetwork) =>
+          multichainNetwork === networkChainId ||
+          multichainNetwork === hexChainId,
+      );
+
+      return isBuiltInNetwork || isFeaturedRpc || isMultichainProviderConfig;
+    },
+    [],
+  );
+
+  return {
+    t,
+    dispatch,
+    trackEvent,
+    hasAnyAccountsInNetwork,
+    history,
+    isNetworkInDefaultNetworkTab,
+
+    // Selectors
+    tokenNetworkFilter,
+    showTestnets,
+    selectedTabOrigin,
+    isUnlocked,
+    domains,
+    orderedNetworksList,
+    customNetworks,
+    isAddingNewNetwork,
+    isMultiRpcOnboarding,
+    isAccessedFromDappConnectedSitePopover,
+    completedOnboarding,
+    isNetworkDiscoverButtonEnabled,
+    multichainNetworks,
+    evmNetworks,
+    currentChainId,
+    enabledNetworksByNamespace,
+    editingChainId,
+    editCompleted,
+    permittedChainIds,
+    permittedAccountAddresses,
+    allChainIds,
+    nonTestNetworks,
+    testNetworks,
```

Comment:
I'm not a huge fan of exporting the selectors themselves in a hook, whichever component is consuming this hook can easily call these selectors itself.

Reply:
As we've not measured performance on this, I don't want it to be a blocker, but it is worth keeping an eye on this.

We should strive hooks and selectors to be hyper-specific and be as minimal as can be.

Reply:
I am happy to refactor this in a fast follow. Maybe we can create a couple of issues around this to keep track of everything.